### PR TITLE
crystal: add fish completions

### PR DESCRIPTION
**Pull request by kachick**
fish completion was introduced in version 1.6.0
so this guard is required, this package is also supporting 1.2.2

https://github.com/crystal-lang/crystal/commit/98a15ee1ab90c2590d43529ea527c6ffc0ce318d
https://github.com/crystal-lang/crystal/blob/0e80a60f0826f6e5031edbd55ee3c3edd9891c18/Makefile#L183-L184

```console
# pkgs.crystal (crystal_1_14)
> tree ./result/share/*sh*
./result/share/bash-completion
└── completions
    └── crystal
./result/share/fish
└── vendor_completions.d
    └── crystal.fish
./result/share/zsh
└── site-functions
    └── _crystal
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/compilers/crystal/default.nix [hunk 1]
```diff
@@ -220,6 +220,8 @@ let
 
         install -Dm644 etc/completion.bash $out/share/bash-completion/completions/crystal
         install -Dm644 etc/completion.zsh $out/share/zsh/site-functions/_crystal
+        # fish completion was introduced since 1.6.0
+        test -f etc/completion.fish && install -Dm644 etc/completion.fish $out/share/fish/vendor_completions.d/crystal.fish
```
Comment (Aleksanaa) [inline pkgs/development/compilers/crystal/default.nix:224]: use `installShellCompletion` for these.

File: pkgs/development/compilers/crystal/default.nix [hunk 2]
```diff
@@ -220,6 +220,8 @@ let
...
 
         install -Dm644 man/crystal.1 $out/share/man/man1/crystal.1
```
Comment (Aleksanaa) [inline pkgs/development/compilers/crystal/default.nix:226]: `installManPage`

**Comment by kachick** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>tijolo</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 39 packages built:</summary>
  <ul>
    <li>ameba</li>
    <li>amqpcat</li>
    <li>blahaj</li>
    <li>collision</li>
    <li>crystal</li>
    <li>crystal.bin</li>
    <li>crystal.lib</li>
    <li>crystal2nix</li>
    <li>crystal_1_11</li>
    <li>crystal_1_11.bin</li>
    <li>crystal_1_11.lib</li>
    <li>crystal_1_12</li>
    <li>crystal_1_12.bin</li>
    <li>crystal_1_12.lib</li>
    <li>crystal_1_2</li>
    <li>crystal_1_2.bin</li>
    <li>crystal_1_2.lib</li>
    <li>crystal_1_7</li>
    <li>crystal_1_7.bin</li>
    <li>crystal_1_7.lib</li>
    <li>crystal_1_8</li>
    <li>crystal_1_8.bin</li>
    <li>crystal_1_8.lib</li>
    <li>crystal_1_9</li>
    <li>crystal_1_9.bin</li>
    <li>crystal_1_9.lib</li>
    <li>crystalline</li>
    <li>gi-crystal</li>
    <li>icr</li>
    <li>invidious</li>
    <li>kakoune-cr</li>
    <li>lucky-cli</li>
    <li>mint</li>
    <li>oq</li>
    <li>rtfm</li>
    <li>scry</li>
    <li>shards</li>
    <li>thicket</li>
    <li>tmuxPlugins.fingers</li>
  </ul>
</details>

File: pkgs/development/compilers/crystal/default.nix (hunk 1 above)
Comment (kachick) [reply pkgs/development/compilers/crystal/default.nix]: Simply replacing with `installShellCompletion` drops current files from `pkgs.crystal`. So added 
 `ln` after it. https://github.com/NixOS/nixpkgs/commit/7ad1c658a7057813b1058aa33c8e2d9f74604da8#diff-1205fbe1629d11896d8a36ee90fed97d79b6dbdf568bf4201483cdcf62a88318R230-R233

**Comment by Aleksanaa** [comment]
@ofborg build crystal

**Pull request merged**